### PR TITLE
Time interval to query parameter

### PR DESCRIPTION
**Pull request by Platayo**
What's New?
-implemented the start and end time interval on the mono query

### CODE REVIEW

File: MLModel.py [hunk 1]
```diff
@@ -5,8 +5,23 @@
 from sklearn.feature_extraction.text import TfidfVectorizer
 from sklearn.cluster import KMeans
 import re
+
+def get_date():
```
Comment (marxprop) [inline MLModel.py:9]: Change this function name to get_transaction_date_range instead.

File: MLModel.py [hunk 2]
```diff
@@ -5,8 +5,23 @@
...
+    end_month = datetime.datetime.now().strftime("%m")
+    end_day = datetime.datetime.now().strftime("%d")
+    end_year = datetime.datetime.now().strftime("%Y")
+    start_year = int(end_year) - 1
+    start_date = datetime.datetime(int(start_year), int(end_month), int(end_day))
+    start_date = start_date.strftime("%d-%m-%Y")
+    return start_date, end_date
+
+start = get_date()[0]
+end = get_date()[1]
```
Comment (marxprop) [inline MLModel.py:20]: Ensure you call the function just once and inside the get transactions function instead (not outside)
Start and End are very vague variable names too. You might want to use clearer variable names.

`start_date, end_date = get_transaction_date_range()`

File: MLModel.py [hunk 3]
```diff
@@ -5,8 +5,23 @@
...
+
+def get_date():
+    end_date = datetime.datetime.now().strftime("%d-%m-%Y")
+    end_month = datetime.datetime.now().strftime("%m")
+    end_day = datetime.datetime.now().strftime("%d")
+    end_year = datetime.datetime.now().strftime("%Y")
+    start_year = int(end_year) - 1
+    start_date = datetime.datetime(int(start_year), int(end_month), int(end_day))
+    start_date = start_date.strftime("%d-%m-%Y")
+    return start_date, end_date
```
Comment (marxprop) [inline MLModel.py:17]: You might want to update this function logic to be more simpler.
```
def get_transaction_date_range():
    current_date = datetime.now()
    end_date = current_date.strftime("%d-%m-%Y")
    start_date = (current_date - timedelta(days=365)).strftime("%d-%m-%Y")
    return start_date, end_date
```

Comment (marxprop) [reply MLModel.py:17]: Please test this see if it does the same thing.

**Review by marxprop — CHANGES_REQUESTED** [review]

**Comment by Platayo** [comment]
I affected the changes. The code you provided worked.

File: MLModel.py [hunk 4]
```diff
@@ -5,8 +5,19 @@
...
 import re
+from datetime import datetime, timedelta
+
+def get_transaction_date_range():
+    current_date = datetime.now()
+    end_date = current_date.strftime("%d-%m-%Y")
+    start_date = (current_date - timedelta(days=365)).strftime("%d-%m-%Y")
+    return start_date, end_date
+
+start_date, end_date = get_transaction_date_range()
```
Comment (marxprop) [inline MLModel.py:16]: This should be inside in the `get_transactions` function instead so it is only used when calling the get_transactions function.

**Review by marxprop — CHANGES_REQUESTED** [review]

**Comment by marxprop** [comment]
@Platayo, Other things to look at
- Once you've fixed a requested change, you can press the `Resolve conservation` button for each comment.
- At the right hand pane, under the Development section, please select the issue this PR is resolving and link it.

**Review by marxprop — APPROVED** [review]

File: MLModel.py [hunk 5]
```diff
@@ -5,8 +5,17 @@
...
+def get_transaction_date_range():
+    current_date = datetime.now()
+    end_date = current_date.strftime("%d-%m-%Y")
+    start_date = (current_date - timedelta(days=365)).strftime("%d-%m-%Y")
+    return start_date, end_date
+
 def get_transactions(account_id):
-    url = f"https://api.withmono.com/accounts/{account_id}/transactions?paginate=false"
+    start_date, end_date = get_transaction_date_range()
+    url = f"https://api.withmono.com/accounts/{account_id}/transactions?paginate=false&start_date&end_date"
```
Comment (marxprop) [inline MLModel.py:18]: Ensure that start_date and end_date are in parentheses.
Please open another PR for that. Missed that before merging
@Platayo

**Pull request merged**